### DISCUSSION TITLE
React v19

### DISCUSSION THREAD

**Discussion by LupusMKW** [reaviz/reagraph#298] [Q&A]
Hi guys,

renovate update me to React v19 but it seems that reagraph currently does not work with React v19. I get the following.

Is this already known? Is there a workaround, solution?
ncaught (in promise) TypeError: ReactSharedInternals is undefined
    $$$reconciler webpack://workcatalog-alt-ui/./node_modules/reagraph/node_modules/@react-three/fiber/node_modules/react-reconciler/cjs/react-reconciler.development.js?:498
    createRenderer webpack://workcatalog-alt-ui/./node_modules/reagraph/node_modules/@react-three/fiber/dist/index-27578f8f.esm.js?:253
    <anonymous> webpack://workcatalog-alt-ui/./node_modules/reagraph/node_modules/@react-three/fiber/dist/index-27578f8f.esm.js?:1730
    node_modules react-three/fiber/dist/index-27578f8f.esm.js@http://localhost:3007/vendors-node_modules_reagraph_node_modules_react-spring_three_dist_react-spring-three_esm_js--ac9003.js:58
    __webpack_require__ http://localhost:3007/main.js:337
    fn http://localhost:3007/main.js:629
    <anonymous> webpack://workcatalog-alt-ui/./node_modules/reagraph/dist/index.js?:76

**Comment by kierancap** [comment]
There are a few dependencies waiting to be updated for React 19 Support.

`@react-three/fiber` has an RC that works with React 19: https://github.com/pmndrs/react-three-fiber/issues/3398#issuecomment-2494653767
`@react-spring/three` is still in progress: https://github.com/pmndrs/react-spring/issues/2341#issuecomment-2528509194

Hopefully once those are updated, Reagraph can support React 19.

**Reply by amcdnl** [reply]
Yes, once the other depedencies support it we can upgrade. Its absolutely on the roadmap!

**Reply by m4ttheweric** [reply]
These have alpha versions that support React 19, could you do an alpha release of reagraph as well?

**Reply by amcdnl** [reply]
@m4ttheweric - Can you make a PR w/ the upgrades, happy to cut a release!~
